Fixes to navigation and safety calculation.

mrguilima opened the pull request:
Fixes to navigation and safety calculation.  The new method updates the navigation state at the post-step point, while the replaced method did not.

Second commit has temporary status of Killed for tracks exiting the geometry setup, until the ExitingSetup status is properly handled.

mrguilima committed:
Fixes to navigation and safety calculation

mrguilima committed:
Temporarily set status=Killed for tracks exiting geometry setup

pcanal commented on source/Geant/track/TrackState.hpp:
```diff
@@ -216,7 +216,8 @@ void LinearStep(TrackState &state, double step)
     Volume_t const* vol = state.fGeometryState.fVolume;
     if (vol) {
-      vol->GetSafetyEstimator()->ComputeSafety(state.fPos, *(state.fGeometryState.fPath));
+      double safety = vol->GetSafetyEstimator()->ComputeSafety(state.fPos, *(state.fGeometryState.fPath));
```
```suggestion
      const double safety = vol->GetSafetyEstimator()->ComputeSafety(state.fPos, *(state.fGeometryState.fPath));
```

mrguilima committed:
Only minor improvements to comments

mrguilima committed:
Add fTotalLength variable to track, and update fNsteps counter after each linear step

mrguilima committed:
Magnetic field simplifications

- Simplify B-field to (0,0,Bz)
- Avoid unnecessary temporary conversion in Vector3D -> ThreeVector

mrguilima committed:
ProxyScattering::AlongStepDoIt() should modify direction only slightly, not too much.

mrguilima committed:
Add more gammas for debugging

mrguilima committed:
Split ALongStep and PostStep GPILs for closer monitoring

mrguilima committed:
Improve identification of tracks at initialization

mrguilima committed:
Minor changes, mostly comments

mrguilima committed:
Add a final track report at last step.

mrguilima committed:
Use simpler geometry dimensions for easier debugging

mrguilima committed:
Minor cleanup in track initialization

mrguilima committed:
Minor fix to class documentation

mrguilima committed:
Remove Transportation from list of processes to be applied

mrguilima committed:
Fixes to TrackState::fTotalLenght

mrguilima committed:
GL: temporary GPU fixes to Soon's change to navigationState allocation

mrguilima commented:
This PR was merged and not closed.  Closing it now.
